src/op_handlers/far_call.rs

```diff
@@ -240,7 +239,9 @@ pub fn far_call(
         .checked_add(stipend)
         .expect("stipend must not cause overflow");
 
-    let program_code = contract_storage
+    let program_code = world
+        .contracts_storage
+        .borrow_mut()
```

Comment:
Please put this into a method, I know it can be a bit more bureaucratic, but this can panic if for some reason the reference holds from another place in the codebase, putting this code inside a method guarantees the reference is dropped. Furthermore, we only want to use the storage for decommit, so this could be simply a 'decommit' method and be done with it.

Reply:
You will also probably have to clone the vec with the program code, I think the cost is worth the hassle since the alternative is panicking.